src/cosalib/meta.py

```diff
@@ -36,55 +67,15 @@ def read(self):
         self.clear()
         # Load the file
         self.update(load_json(self._meta_path))
+        self.validate()
 
     def write(self):
         """
         Write out the dict to the meta path.
         """
+        self.validate()
```

Comment:
Seems odd to verify the *previous* version here - I think we want to do something like write to a temp file, verify that, then `rename()` ?

Reply:
My thinking here is that we prevent the writing of an invalid `meta.json`. We do validate on `read` to make sure that the starting `meta.json` but as classes updated it (ala `cosalib.qemuvariants` and `cosalib.{azure,aliyun,aws,gcp}.py` their updates would be validated. 

Call `self.validate()` checks the in-memory `dict(self)` before we commit it to disk.